ballerina/annotation.bal

```diff
@@ -14,9 +14,12 @@
 // specific language governing permissions and limitations
 // under the License.
 
+import ballerina/jballerina.java;
+
 # The record definition for the Descriptor annotation
 #
 # + value - The descriptor value as a hexadecimal string
+@java:ExternalDependency
 public type MessageDescriptor record {|
```

Comment:
This is actually the type of the annotation and the relevant annotation is present in the test source. In that case this type should be reachable right? Do we really need to add the external dependency annotation?
Created an issue here: https://github.com/ballerina-platform/ballerina-lang/issues/43580

Reply:
I dont think we can use `@ExternalDependency` here. Only type dependencies from java side should be annotated with `@ExternalDependency`. Since this type is referenced directly in a ballerina to ballerina way this should be picked up and marked as "used" by the DCE analyser. Must be a bug when analysing annotations in BIR level. 

We will have to fix this from the compiler side.

Reply:
Ack 👍🏼

Reply:
Removed via [1d57089](https://github.com/ballerina-platform/module-ballerina-protobuf/pull/245/commits/1d57089a11783f044746dce09b0d9f3c33d8f491)